pkg/api/v2/reporter/reporter.go

```diff
@@ -165,11 +165,15 @@ func (r *reporter) WriteReports(ctx context.Context, resourceErrs ResourceReport
 }
 
 func attemptUpdateStatus(ctx context.Context, client clients.ResourceClient, resourceToWrite resources.InputResource) (resources.Resource, resources.InputResource, error) {
+	var readErr error
+	_, readErr = client.Read(resourceToWrite.GetMetadata().Namespace, resourceToWrite.GetMetadata().Name, clients.ReadOpts{Ctx: ctx})
+	if readErr != nil { // resource has been deleted, don't re-create
+		return resourceToWrite, resourceToWrite, nil
```

Comment:
should be
```suggestion
		return nil, resourceToWrite, nil
```
as the return args are `// updatedResource, resourceToWrite, error`

Reply:
and then we need to wrap
```golang
		resources.UpdateMetadata(resource, func(meta *core.Metadata) {
			meta.ResourceVersion = updatedResource.GetMetadata().ResourceVersion
		})

		logger.Debugf("wrote report %v : %v", updatedResource.GetMetadata().Ref(), status)
```
in a nil check on `updatedResource`